src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Xamarin.Android.Sdk.props

```diff
@@ -29,6 +29,12 @@ Copyright (C) 2011-2016 Xamarin. All rights reserved.
 		<FrameworkPathOverride>$(TargetFrameworkRootPath)MonoAndroid\v1.0</FrameworkPathOverride>
 	</PropertyGroup>
 
+	<Target Name="DesignTimeSetup">
+		<PropertyGroup>
+			<DesignTimeBuild Condition=" '$(DesignTimeBuild)' == '' And '$(BuildingInsideVisualStudio)'=='true' And '$(BuildingOutOfProcess)'=='false' ">true</DesignTimeBuild>
```

Comment:
To add to the thread: this is the "source of inspiration": https://github.com/Microsoft/msbuild/commit/5262c51b7b8799c3d65680776c866d5c356632bb